lib/jit_preloader/active_record/associations/preloader/ar7_branch.rb

```diff
@@ -0,0 +1,24 @@
+module JitPreloader
+  module PreloaderBranch
+    """
+    From ActiveRecord version >= 7.x.x, it introduces a improvement for preloading associations in batch
+    https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader.rb#L121
+
+    Previously we make sure to check and direct return if an association klass is not a valid ActiveRecord::Base
+    class through prepended methods within lib/jit_preloader/active_record/associations/preloader/arx_association.rb
+    The Batch approach move the check of association klass forward before calling individual association.
+    https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader/batch.rb#L17
+
+    This module prepend on Branch class which is the object we grab the list of loaders aka associations
+    and remove the one that does not have a valid ActiveRecord::Base klass
+    """
+
+    def loaders
+      @loaders = super.reject do |loader|
+        !loader.klass.respond_to?(:base_class)
```

Comment:
Remove the negative?
```suggestion
      @loaders = super.find_all do |loader|
        loader.klass.respond_to?(:base_class)
```